### PR TITLE
21025 create configuration endpoint

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#21025

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/models/configuration.py [hunk 1]
```diff
@@ -64,6 +82,12 @@ def validate_value(self):
                 int(self.val)
             except ValueError as exc:
                 raise ValueError(f'Value for key {self.name} must be an integer') from exc
+            num_dissolutions_allowed = self.val if self.name == 'NUM_DISSOLUTIONS_ALLOWED'\
+                else Configuration.find_by_name('NUM_DISSOLUTIONS_ALLOWED').val
+            max_dissolutions_allowed = self.val if self.name == 'MAX_DISSOLUTIONS_ALLOWED'\
+                else Configuration.find_by_name('MAX_DISSOLUTIONS_ALLOWED').val
+            if int(num_dissolutions_allowed) > int(max_dissolutions_allowed):
+                raise ValueError('NUM_DISSOLUTIONS_ALLOWED cannot be greater than MAX_DISSOLUTIONS_ALLOWED.')
```
Comment (argush3) [inline legal-api/src/legal_api/models/configuration.py:90]: I don't think you should be doing this type of validation here.

We should do it further up where the resource is called.

File: legal-api/src/legal_api/resources/v2/configuration.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return ({'message': 'Request body cannot be blank'}), HTTPStatus.BAD_REQUEST
+
+    configurations = json_input['configurations']
+
+    for data in configurations:
+        if data['name']:
+            configuration = Configuration.find_by_name(data['name'])
+            configuration.val = data.get('value', configuration.val)
+            configuration.full_description = data.get('full_description', configuration.full_description)
+            configuration.short_description = data.get('short_description', configuration.short_description)
```
Comment (argush3) [inline legal-api/src/legal_api/resources/v2/configuration.py:57]: for now we only support updating the value.  you can remove the description fields.

File: legal-api/src/legal_api/resources/v2/configuration.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        'configurations': [
+            configuration.json for configuration in configurations
+        ]
+    }), HTTPStatus.OK
+
+
+@bp.route('', methods=['PUT'])
+@cross_origin(origin='*')
+@jwt.has_one_of_roles([UserRoles.staff])
+def update_configurations():
```
Comment (argush3) [inline legal-api/src/legal_api/resources/v2/configuration.py:43]: can we put the core logic of this endpoint into a service class?

Comment (argush3) [inline legal-api/src/legal_api/resources/v2/configuration.py:43]: you'll need to make sure you have duplicate key validation and any other validation you haven't addressed that was mentioned in the ticket desc.

File: legal-api/src/legal_api/resources/v2/configuration.py (hunk 2 above)
Comment (kzdev420) [reply legal-api/src/legal_api/resources/v2/configuration.py:57]: Removed

File: legal-api/src/legal_api/models/configuration.py (hunk 1 above)
Comment (kzdev420) [reply legal-api/src/legal_api/models/configuration.py:90]: Updated

**Comment by argush3** [comment]
@kzdev420 this PR can be closed right?

**Comment by kzdev420** [comment]
> @kzdev420 this PR can be closed right?

Yes, This can be close

**Pull request closed without merging**